zulipterminal/model.py

```diff
@@ -198,17 +198,23 @@ def _notify_server_of_presence(self) -> None:
                 }
             )
         self.new_user_input = False
+        return response
 
     @asynch
     def _start_presence_updates(self) -> None:
         """
-        Notify server of user's presence each minute (version 1a)
+        Call `_notify_server_of_presence` every minute (version 1a).
+        Use 'response' to update user list (version 1b).
         """
-        # FIXME: version 1b: Also use 'response' to update user list
         # FIXME: Version 2: call endpoint with ping_only=True only when
         #        needed, and rely on presence events to update
         while True:
-            self._notify_server_of_presence()
+            response = self._notify_server_of_presence()
+            self.initial_data['presences'] = response['presences']
```

Comment:
@sumanthvrao response object is not guaranteed to have `presences`. Also, ` self._notify_server_of_presence()` should be in try catch statement.

Reply:
Ah. All-right. Makes sence. Will do the additions :+1:

Reply:
This may also be the issue @neiljp was facing (KeyError?)

Reply:
Oops, `run.py` will catch error in `self._notify_server_of_presence()` , so no need to put in try-catch statement. ;)

Reply:
we should always check for `if response['result'] == 'success':`

Reply:
Ah. Got it. Yup, am adding the response check for success.

Reply:
@amanagr Good review catch in the checking for success here - I've not reproduced the issue I had previously with this PR, in this updated version :+1:  I'm not convinced that the try/catch would help here, as `_start_presence_updates` is `@asynch`, so the exceptions are in different threads, and that doesn't just resolve itself automatically (as possibly previously discussed elsewhere).